### PR TITLE
codec: support mysql Duration codec

### PR DESCRIPTION
**Pull request by siddontang**

### CODE REVIEW

File: util/codec/codec.go [hunk 1]
```diff
@@ -36,6 +38,7 @@ const (
 	formatFloatFlag  = 'f'
 	formatStringFlag = 's'
 	formatBytesFlag  = 'b'
+	formatDuration   = 't'
```
Comment (qiuyesuifeng) [inline util/codec/codec.go]: formatDuration -> formatDurationFlag?

Comment (siddontang) [inline util/codec/codec.go]: yes, my fault.

File: util/codec/codec.go [hunk 2]
```diff
@@ -163,6 +170,13 @@ func DecodeKey(b []byte) ([]interface{}, error) {
 			}
 		case formatBytesFlag:
 			b, v[i], err = DecodeBytes(b)
+		case formatDurationFlag:
+			var r int64
+			b, r, err = DecodeInt(b)
+			if err == nil {
+				// use max fsp, let outer to do round manually.
+				v[i] = mysql.Duration{Duration: time.Duration(r), Fsp: mysql.MaxFsp}
```
Comment (shenli) [inline util/codec/codec.go]: How about encode fsp and decode it here?

Comment (siddontang) [inline util/codec/codec.go]: maybe no need here, but I am not sure. 
As you know, we handle encode any fsp in table value codec or time marshal too, just let outer do the convert manually.

**Comment by shenli** [comment]
LGTM

**Pull request merged**